src/hotspot/os_cpu/linux_x86/os_linux_x86.cpp

```diff
@@ -544,6 +544,21 @@ void os::print_context(outputStream *st, const void *context) {
   st->print(", ERR=" INTPTR_FORMAT, (intptr_t)uc->uc_mcontext.gregs[REG_ERR]);
   st->cr();
   st->print("  TRAPNO=" INTPTR_FORMAT, (intptr_t)uc->uc_mcontext.gregs[REG_TRAPNO]);
+  // Add XMM registers + MXCSR. Note that C2 uses XMM to spill GPR values including pointers.
+  st->cr();
+  st->cr();
+  size_t fpregs_offset = ((address)uc->uc_mcontext.fpregs >= (address)uc) ?
+                         pointer_delta(uc->uc_mcontext.fpregs, uc, 1) : 0;
+  if (fpregs_offset >= sizeof(ucontext_t) || fpregs_offset == 0) {
```

Comment:
Why protect against the assertion in `pointer_delta` here and not also in `store_context`?
You could do the following in both methods:
```suggestion
  size_t fpregs_offset = ((address)uc->uc_mcontext.fpregs >= (address)uc) ?
                         pointer_delta(uc->uc_mcontext.fpregs, uc, 1) : sizeof(ucontext_t);
  if (fpregs_offset >= sizeof(ucontext_t)) {
```

Reply:
I've changed it such that we don't touch in this case.